source/max.js

```diff
@@ -18,5 +19,9 @@ import _curry2 from './internal/_curry2.js';
  *      R.max(789, 123); //=> 789
  *      R.max('a', 'b'); //=> 'b'
  */
-var max = _curry2(function max(a, b) { return b > a ? b : a; });
+var max = _curry2(function max(a, b) {
+  if (equals(a, b)) { return b; }
+  if (a > b || b > a) { return b > a ? b : a; }
+  throw new TypeError('cannot compare ' + toString(a) + ' with ' + toString(b));
```

Comment:
I'm not too sure about this.

I would expect people to use `maxBy` or `minBy` for values that can't be compared directly e.g.

```javascript
maxBy(path(['a','b','c']))({a:{b:{c:42}}})({a:{b:{c:41}}})
```

I understand your motivations (I think) but unexpected result is what you get when you try to compare apples and pears. If I follow this change correctly then:

```javascript
max([10,2])([10,10])
//=> [10,2]
```

I'm not sure this is a result that most people would understand.

What do you think?

Reply:
My use case for adding this error handling was based on when we can't predict with certainty what the values will be. For example, we may expect values, but receive `undefined` - this will always result in the first argument being returned by `min`/`max`.

```
min(undefined, 1) // => undefined
min(1, undefined) // => 1
```

This may not be immediately obvious when using the functions, and cause some strange outcomes.

That said, perhaps a lighter approach may be better, where we assume the correct _type_ will be passed and only check that both args are defined.

Reply:
> That said, perhaps a lighter approach may be better, where we assume the correct type will be passed and only check that both args are defined.

So if both `a` and `b` can't be compared and one is nil returned the other?

Sounds reasonable to me but maybe I'm overlooking something @ramda/core ?